madras-simulator/MADRaS
# Gym integration

What?
- Adds support for the MADRaS Env (`Madras-v0`)
- Creates a proper python package for the repo
   - Added `setup.py`
   - Changed Repo structure
   - Fixed typo in `__init__.py` filenames

Why?
To remove the dependency of keeping a local installation of the full gym repo [gym-env](https://github.com/madras-simulator/gym/pull/1).
 
Installation:
```shell
pip install -e .
```
Testing instructions:
Keep the sim in running condition t.e. `sh startTorcs.sh`

For testing in python shell
```python
>>> import MADRaS
>>> import gym
>>> env = gym.make('Madras-v0')
>>> env.reset()
>>> env.step(env.action_space.sample())
```
For testing with baselines:
A slight modification of the baselines code is req:
similar to the pr [baselines](https://github.com/madras-simulator/baselines/pull/1)
- Add two lines in `baselines/run.py` 
```python
import MADRAS # at the top
_game_envs['madras'] = {'Madras-v0'} #line 52
``` 
- Comment out line54 `assert (np.abs(env.action_space.low) == env.action_space.high).all()  # we assume symmetric actions.` from `/baselines/ddpg/ddpg.py`

NOTE: 
This PR still uses the current version of `gym_madras.py` not the updated one from [gym](https://github.com/madras-simulator/gym/pull/1) pr.

I have tested with baselines and although it seems that the training is occurring i.e. the agent is running and reset everything is working fine but I cannot see any output in the terminal. Also, I tested the older pr #7 which also seemed to have this issue.
